### PR TITLE
Feature/restore folder

### PR DESCRIPTION
**Pull request by valer-kovac**
RestoreTrashedFolderAsync - Updated validation of input parameters

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @valer-kovac has signed the CLA. Thanks for the pull request!

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 1]
```diff
@@ -189,6 +189,7 @@ public BoxFilesManagerTest()
 
             BoxFileRequest request = new BoxFileRequest()
             {
+                Id="5000948880",
```
Comment (coolboy) [inline Box.V2.Test/BoxFilesManagerTest.cs]: space between =

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 2]
```diff
@@ -264,19 +264,32 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
...
                 .Id.ThrowIfNullOrWhiteSpace("folderRequest.Parent.Id");
 
             BoxRequest request = new BoxRequest(_config.FoldersEndpointUri, folderRequest.Id)
                     .Method(RequestMethod.Post)
-                    .Param(ParamFields, fields)
-                    .Payload(_converter.Serialize(folderRequest));
+                    .Param(ParamFields, fields);
+
+            // ID shall not be used in request body it is used only as url attribute
+            string oldId = folderRequest.Id;
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: I am not fully understand why you are going the resetting here, could you explain?

Comment (valer-kovac) [inline Box.V2/Managers/BoxFoldersManager.cs]: I decided to rememeber and renew ID value after processing to let it be transparent from caller point of view. I think it is not very good to change value of outside data internally may influence caller in some cases if he decides to use this value somehow again after method call.

**Comment by coolboy** [comment]
@marek-vysoky conflict

**Comment by marek-vysoky** [comment]
Conflict resolved

**Comment by cburnette** [comment]
@coolboy Needs review

**Pull request merged**